### PR TITLE
Some updates on shell-host.c

### PR DESCRIPTION
**Pull request by bingbing8**
https://github.com/PowerShell/Win32-OpenSSH/issues/893
Create pipe using default buffer size
Update buffer size when read and write file
Change to use _wcsnicmp.

### CODE REVIEW

File: contrib/win32/win32compat/shell-host.c [hunk 1]
```diff
@@ -1288,7 +1289,8 @@ start_withno_pty(wchar_t *command)
 
 	memset(&sa, 0, sizeof(SECURITY_ATTRIBUTES));
 	sa.bInheritHandle = TRUE;
-	if (!CreatePipe(&child_pipe_read, &child_pipe_write, &sa, 128))
+	/* use the default buffer size, 64K*/
+	if (!CreatePipe(&child_pipe_read, &child_pipe_write, &sa, 0))
 		return -1;
```
Comment (bagajjal) [inline contrib/win32/win32compat/shell-host.c]: Please add a printf with the GetLastError()

File: contrib/win32/win32compat/shell-host.c [hunk 2]
```diff
@@ -1321,16 +1323,16 @@ start_withno_pty(wchar_t *command)
...
+		    (command_len >= 7 && _wcsnicmp(command, L"cmd.exe", 8) == 0) ||
+		    (command_len >= 4 && _wcsnicmp(command, L"cmd ", 4) == 0) ||
+		    (command_len >= 8 && _wcsnicmp(command, L"cmd.exe ", 8) == 0))
 			process_input = TRUE;
 	}
 
 	/* Try launching command as is first */
-	if (command) {
-		ret = CreateProcessW(NULL, command, NULL, NULL, TRUE, 0, NULL, NULL, &si, &pi);
+	if (command) {
```
Comment (bagajjal) [inline contrib/win32/win32compat/shell-host.c]: remove the extra spaces like this (after { ).. 
There are multiple places like this.

**Review by bagajjal — APPROVED** [review]

File: contrib/win32/win32compat/shell-host.c [hunk 3]
```diff
@@ -1105,18 +1105,13 @@ start_with_pty(wchar_t *command)
 {
 	STARTUPINFO si;
 	PROCESS_INFORMATION pi;
-	wchar_t *cmd = (wchar_t *)malloc(sizeof(wchar_t) * MAX_CMD_LEN);
+	wchar_t cmd[MAX_CMD_LEN] = {L'\0',};
```
Comment (bagajjal) [inline contrib/win32/win32compat/shell-host.c]: I made this change to fix warning message.. having this as non-static memory will not cause any issue?

Comment (bingbing8) [reply contrib/win32/win32compat/shell-host.c]: don't see any warning

**Pull request merged**